migrations/app/schema/20241119151019_stored_procs_for_ordering_service_items.up.sql

```diff
@@ -0,0 +1,365 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+ALTER TABLE mto_service_items
+  ALTER COLUMN id SET DEFAULT uuid_generate_v4();
+
+
+-- creating function to get address is_oconus
+CREATE OR REPLACE FUNCTION get_is_oconus(address_id UUID)
+RETURNS BOOLEAN AS $$
+DECLARE
+    is_oconus BOOLEAN;
+BEGIN
+    SELECT a.is_oconus
+    INTO is_oconus
+    FROM addresses a
+    WHERE a.id = address_id;
+
+    RETURN is_oconus;
+EXCEPTION
+    WHEN NO_DATA_FOUND THEN
+        RAISE EXCEPTION 'Address with ID % not found', address_id;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION checkExistingServiceItem(
+    service_id UUID,
+    shipment_id UUID
+) RETURNS BOOLEAN AS $$
+BEGIN
+    IF EXISTS (
+        SELECT 1
+        FROM mto_service_items
+        WHERE re_service_id = service_id
+        AND mto_shipment_id = shipment_id
+    ) THEN
+        RAISE EXCEPTION 'Service item already exists for service_id % and shipment_id %', service_id, shipment_id;
+    END IF;
+    RETURN FALSE;
+END;
+$$ LANGUAGE plpgsql;
+
+
+-- stored proc that creates auto-approved service items based off of a shipment id
+CREATE OR REPLACE PROCEDURE CreateApprovedServiceItemsForShipment(
+    IN shipment_id UUID
+)
+AS '
+DECLARE
+    s_status mto_shipment_status;
+    s_status mto_shipment_status;
```

Comment:
```suggestion
    s_status mto_shipment_status;
```
Duplicate needs removal

Reply:
fixed